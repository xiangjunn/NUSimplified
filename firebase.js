import * as firebase from 'firebase';
import '@firebase/auth';
import '@firebase/firestore';
import {API_KEY, PROJECT_NAME, MESSAGING_SENDER_ID, APP_ID} from '@env'

const firebaseConfig = {
  apiKey: API_KEY,
  authDomain: PROJECT_NAME + '.firebaseapp.com',
  databaseURL: 'https://' + PROJECT_NAME +'.firebaseio.com',
  projectId: PROJECT_NAME,
  storageBucket: PROJECT_NAME + '.appspot.com',
  messagingSenderId: MESSAGING_SENDER_ID,
  appId: APP_ID,
};

if (!firebase.apps.length) {
    firebase.initializeApp(firebaseConfig);
}

export { firebase };