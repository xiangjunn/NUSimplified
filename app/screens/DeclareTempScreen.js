import { Container, Content, Button, Text, Form } from 'native-base';
import React, { useState } from 'react';
import { StyleSheet } from 'react-native';
import Slider from '@react-native-community/slider';
import { firebase } from '../../firebase'


function DeclareTempScreen() {
  const [temperature, setTemperature] = useState(370); // multiplied by 10 because slider cannot work properly with decimal
  const [success, setSuccess] = useState(false);
  const userId = firebase.auth().currentUser.uid;
  const db = firebase.firestore();

  async function submitDeclaration() {
    const currentTime = firebase.firestore.Timestamp.now().toDate();
    const isNormal = temperature < 375;
    const actualTemp = temperature / 10;
    const tempInfo = {
      timestamp: currentTime,
      temperature: actualTemp,
      isNormal
    }
    await db.collection("users").doc(userId).update({
      temperatureHistory: firebase.firestore.FieldValue.arrayUnion(tempInfo)
    })
    setSuccess(true);
  }

    return (
        <Container >
          <Content>
            <Text style={{fontSize: 100, textAlign: 'center', marginTop: 50}}>{(temperature / 10).toFixed(1)}</Text>
            <Form style={{flexDirection: 'row', height: 80}}>
              <Form style={{alignItems: 'flex-start', flex: 1, flexDirection: 'row'}}>
            <Button style={styles.button} onPress={() => temperature <= 355 ? setTemperature(350) : setTemperature(temperature - 5)}>
              <Text style={styles.text}>-0.5</Text>
            </Button>
            <Button style={styles.button} onPress={() => temperature <= 351 ? setTemperature(350) : setTemperature(temperature - 1)}>
              <Text style={styles.text}>-0.1</Text>
            </Button>
            </Form>
            <Form style={{justifyContent: 'flex-end', flex: 1, flexDirection: 'row'}}>
            <Button style={styles.button} onPress={() => temperature >= 399 ? setTemperature(400) : setTemperature(temperature + 1)}>
              <Text style={styles.text}>+0.1</Text>
            </Button>
            <Button style={styles.button} onPress={() => temperature >= 395 ? setTemperature(400) : setTemperature(temperature + 5)}>
              <Text style={styles.text}>+0.5</Text>
            </Button>
            </Form>
            </Form>
          <Slider
            style={{width: '80%', height: 40, alignSelf: 'center'}}
            minimumValue={350}
            maximumValue={400}
            value={temperature}
            step={1}
            onValueChange={(temp) => setTemperature(temp)}
            minimumTrackTintColor={temperature < 375 ? "blue" : "red"}
            maximumTrackTintColor={temperature < 375 ? "rgba(0, 0, 255, 0.3)" : "rgba(255, 0, 0, 0.3)"}
           />
            {success
            ? <Form style={{marginHorizontal: 20, marginTop: 100}}>
                <Text style={styles.submitText}>
                Success!
                </Text>
                <Button block disabled>
                  <Text style={styles.submitButton}>Submitted</Text>
                </Button>
              </Form>
            : <Form style={{marginHorizontal: 20, marginTop: 100}}>
                <Text style={styles.submitText}></Text>
                <Button block warning onPress={() => submitDeclaration()}>
                  <Text style={styles.submitButton}>Submit</Text>
                </Button>
              </Form>}
          </Content>
        </Container>
    );
}

const styles = StyleSheet.create({
  button: {
    height: 70,
    margin: 5,
    width: '37%',
    borderRadius: 50
  },
  text: {
    textAlign: 'center',
    width: '100%',
    fontSize: 17
  },
  submitText: {
    textAlign: 'center',
    color: 'green',
    margin: 10,
    fontSize: 15,
    fontWeight: 'bold'
  },
  submitButton: {
    color: 'black',
    fontWeight: 'bold'
  }
});

export default DeclareTempScreen;