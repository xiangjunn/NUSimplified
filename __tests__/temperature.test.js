import React from 'react';
import { fireEvent, render, waitFor, cleanup, act, unmount } from '@testing-library/react-native';
import DeclareTempScreen from '../app/screens/DeclareTempScreen';
import { SAMPLE_EMAIL, SAMPLE_PASSWORD, SAMPLE_EMAIL_UNVERIFIED, SAMPLE_PASSWORD_UNVERIFIED } from '@env'
import { firebase } from '../firebase'

afterEach(() => {    
  jest.clearAllMocks();
  cleanup();
});


window.addEventListener = jest.fn();
window.attachEvent= jest.fn();

beforeAll((done) => {
    firebase.auth().signInWithEmailAndPassword(SAMPLE_EMAIL, SAMPLE_PASSWORD).then(() => done())
})

describe('Declare Temperature Test', () => {
    test('renders correctly', () => {
        const tree = render(<DeclareTempScreen />).toJSON();
        expect(tree).toMatchSnapshot();
    });  

    test('After submitting temperature, there should be a success message', async () => {
        const { getByText } = render(<DeclareTempScreen />);
        const button = getByText('Submit').parent;
        fireEvent.press(button);
        await waitFor(() => expect(getByText('Success!')).toBeDefined());
    })
})

