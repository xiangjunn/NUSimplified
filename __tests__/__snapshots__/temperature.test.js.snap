// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Declare Temperature Test renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "height": 1334,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      automaticallyAdjustContentInsets={false}
      contentContainerStyle={
        Array [
          Object {
            "padding": undefined,
          },
          undefined,
        ]
      }
      contentInset={
        Object {
          "bottom": 0,
        }
      }
      enableAutomaticScroll={true}
      enableOnAndroid={false}
      enableResetScrollToCoords={true}
      extraHeight={75}
      extraScrollHeight={0}
      getScrollResponder={[Function]}
      handleOnScroll={[Function]}
      keyboardDismissMode="interactive"
      keyboardOpeningTime={250}
      keyboardShouldPersistTaps="handled"
      keyboardSpace={0}
      onScroll={[Function]}
      resetKeyboardSpace={[Function]}
      resetScrollToCoords={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      scrollEventThrottle={1}
      scrollForExtraHeightOnAndroid={[Function]}
      scrollIntoView={[Function]}
      scrollToEnd={[Function]}
      scrollToFocusedInput={[Function]}
      scrollToPosition={[Function]}
      showsVerticalScrollIndicator={true}
      style={
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
        }
      }
      update={[Function]}
      viewIsInsideTabBar={false}
    >
      <View>
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              },
              Object {
                "fontSize": 30,
                "marginTop": 50,
                "textAlign": "center",
              },
            ]
          }
          uppercase={false}
        >
          Period: PM
        </Text>
        <TextInput
          allowFontScaling={true}
          editable={true}
          keyboardType="number-pad"
          maxLength={4}
          onChangeText={[Function]}
          placeholderTextColor="#575757"
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "color": "#000",
                "flex": 1,
                "fontSize": 17,
                "height": 50,
                "paddingLeft": 5,
                "paddingRight": 5,
              },
              Object {
                "alignSelf": "center",
                "fontSize": 100,
                "height": 150,
                "marginBottom": 50,
              },
            ]
          }
          underlineColorAndroid="rgba(0,0,0,0)"
          value="37.0"
        />
        <RNCSlider
          disabled={false}
          enabled={true}
          inverted={false}
          maximumTrackTintColor="rgba(0, 0, 255, 0.3)"
          maximumValue={400}
          minimumTrackTintColor="blue"
          minimumValue={350}
          onChange={[Function]}
          onRNCSliderSlidingComplete={null}
          onRNCSliderSlidingStart={null}
          onRNCSliderValueChange={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          step={1}
          style={
            Array [
              Object {
                "height": 40,
              },
              Object {
                "alignSelf": "center",
                "height": 40,
                "width": "80%",
              },
            ]
          }
          thumbImage={
            Object {
              "__packager_asset": true,
              "height": 1,
              "scale": 1,
              "uri": "uri",
              "width": 1,
            }
          }
          value={370}
        />
        <View
          style={
            Array [
              Object {},
              Object {
                "marginHorizontal": 20,
                "marginTop": 100,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                },
                Object {
                  "color": "green",
                  "fontSize": 15,
                  "fontWeight": "bold",
                  "margin": 10,
                  "textAlign": "center",
                },
              ]
            }
            uppercase={false}
          />
          <View
            accessible={true}
            focusable={true}
            forwardedRef={[Function]}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "backgroundColor": "#f0ad4e",
                "borderBottomWidth": null,
                "borderColor": "#007aff",
                "borderLeftWidth": null,
                "borderRadius": 5,
                "borderRightWidth": null,
                "borderTopWidth": null,
                "borderWidth": undefined,
                "elevation": 2,
                "flexDirection": "row",
                "height": 45,
                "justifyContent": "center",
                "opacity": 1,
                "paddingBottom": 6,
                "paddingTop": 6,
                "shadowColor": undefined,
                "shadowOffset": undefined,
                "shadowOpacity": undefined,
                "shadowRadius": undefined,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#fff",
                    "fontFamily": "System",
                    "fontSize": 16.5,
                    "marginLeft": 0,
                    "marginRight": 0,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                  },
                  Object {
                    "color": "black",
                    "fontWeight": "bold",
                  },
                ]
              }
              uppercase={false}
            >
              Submit
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </RCTSafeAreaView>
</View>
`;
